discovery/scaleway/instance.go

```diff
@@ -174,20 +174,24 @@ func (d *instanceDiscovery) refresh(ctx context.Context) ([]*targetgroup.Group,
 			labels[instanceTagsLabel] = model.LabelValue(tags)
 		}
 
+		addr := ""
 		if server.IPv6 != nil {
 			labels[instancePublicIPv6Label] = model.LabelValue(server.IPv6.Address.String())
+			addr = net.JoinHostPort(server.IPv6.Address.String(), strconv.FormatUint(uint64(d.port), 10))
 		}
 
 		if server.PublicIP != nil {
 			labels[instancePublicIPv4Label] = model.LabelValue(server.PublicIP.Address.String())
+			addr = net.JoinHostPort(server.PublicIP.Address.String(), strconv.FormatUint(uint64(d.port), 10))
 		}
 
 		if server.PrivateIP != nil {
 			labels[instancePrivateIPv4Label] = model.LabelValue(*server.PrivateIP)
+			addr = net.JoinHostPort(*server.PrivateIP, strconv.FormatUint(uint64(d.port), 10))
+		}
 
-			addr := net.JoinHostPort(*server.PrivateIP, strconv.FormatUint(uint64(d.port), 10))
+		if addr != "" {
```

Comment:
Thanks.

I'm not familiar with the API, but is it possible for the pointers to be not nil but point to the empty string? If that is possible, then this changes the behavior a little bit, as it would have been possible, before, for `model.AddressLabel` to be set to `""` but now we completely skip setting that label if it's the case, I'm not sure if it's ok.

Adding a test for this would be great especially to enforce the use case and avoid regressions.

I think we'll need to adjust the doc now https://github.com/prometheus/prometheus/blob/b938bbc11169dc224c103bec32a55daad0f3f34e/docs/configuration/configuration.md?plain=1#L2952 to inform that we try other addresses.

Reply:
I think so. Yes, add the case to TestScalewayInstanceRefresh should do it.